File: src/main/scala/s/command/drawer/DrawCanvas.scala
/* (C) 2022 */
package s.command.drawer

import s.command.command.MacroCommand

import java.awt.Canvas
import java.awt.Color
import java.awt.Graphics

class DrawCanvas(
    width: Int,
    height: Int,
    var history: MacroCommand
) extends Canvas
    with Drawable {
  setSize(width, height)
  setBackground(Color.white)
  private val color = Color.red
  private val radius = 6

  override def paint(g: Graphics): Unit = {
    history.execute()
  }

  override def draw(x: Int, y: Int): Unit = {
    val g = getGraphics
    g.setColor(color)
    g.fillOval(x - radius, y - radius, radius * 2, radius * 2)
  }
}

File: src/main/java/j/command/drawer/DrawCanvas.java
package j.command.drawer;

import j.command.command.MacroCommand;

import java.awt.*;

public final class DrawCanvas extends Canvas implements Drawable {
    private final Color color = Color.red;
    private final int radius = 6;
    private final MacroCommand history;

    public DrawCanvas(int width, int height, MacroCommand history) {
        setSize(width, height);
        setBackground(Color.white);
        this.history = history;
    }

    @Override
    public void paint(Graphics g) {
        history.execute();
    }

    @Override
    public void draw(int x, int y) {
        Graphics g = getGraphics();
        g.setColor(color);
        g.fillOval(x - radius, y - radius, radius * 2, radius * 2);
    }
}
